src/data_structures/hashtable.rs

```diff
@@ -0,0 +1,145 @@
+use std::collections::LinkedList;
+
+pub struct HashTable<K, V> {
+    elements: Vec<LinkedList<(K, V)>>,
+    count: usize,
+}
+
+impl<K: Hashable + std::cmp::PartialEq, V> Default for HashTable<K, V> {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+pub trait Hashable {
+    fn hash(&self) -> usize;
+}
+
+impl<K: Hashable + std::cmp::PartialEq, V> HashTable<K, V> {
+    pub fn new() -> HashTable<K, V> {
+        let initial_capacity = 3000;
```

Comment:
nit: Extract to a constant. `const INITIAL_CAPACITY: usize = 3000;`.

Why was this value chosen? `std::collection::HashMap` has an initial capacity of `6` for example.

Reply:
To minimize resizing for larger datasets.